src/ResourceManager/ServiceFabric/Commands.ServiceFabric/Commands/ServiceFabricClusterCmdlet.cs

```diff
@@ -39,6 +42,77 @@ public abstract class ServiceFabricClusterCmdlet : ServiceFabricCmdletBase
 
         #region SFRP
 
+        #region Temporary code until next SDK version is released
+        protected PSCluster SendPutRequest(Cluster clusterResource, bool runOnSameThread = true)
+        {
+            if (runOnSameThread)
+            {
+                WriteVerboseWithTimestamp("Begin to update the cluster");
+            }
+
+            Cluster cluster = null;
+            var tokenSource = new CancellationTokenSource();
+            try
+            {
+                var putRequest = Task.Factory.StartNew(() =>
+                {
+                    try
+                    {
+                        cluster = this.SFRPClient.Clusters.Create(this.ResourceGroupName, this.Name, clusterResource);
+                    }
+                    finally
+                    {
+                        tokenSource.Cancel();
+                    }
+                });
+
+                while (!tokenSource.IsCancellationRequested)
+                {
+                    if (runOnSameThread)
+                    {
+                        if (!RunningTest)
+                        {
+                            var c = SafeGetResource(this.GetCurrentCluster, true);
+                            if (c != null)
+                            {
+                                WriteVerboseWithTimestamp(
+                                    string.Format(
+                                        ServiceFabricProperties.Resources.ClusterStateVerbose,
+                                        c.ClusterState));
+                            }
+                        }
+                    }
+
+                    Thread.Sleep(TimeSpan.FromSeconds(WriteVerboseIntervalInSec));
```

Comment:
@juhacket what is this needed for?

Reply:
This is required for us to change certain properties of the cluster resource. The current API only supports a subset of these properties (SendPatchRequest). We are going to expose these properties to the patch request in the next SDK update. We will merge SendPutRequest into SendPatchRequest in that change as well.